torch/_inductor/freezing.py

```diff
@@ -135,13 +135,35 @@ def run(self):
         return super().run(initial_env=env)
 
 
+def is_impure(node: torch.fx.node.Node):
+    if node.target == torch.ops.quantized_decomposed.dequantize_per_channel.default:
+        # For the pattern fp32_weight -> quantized_decomposed.quantize_per_channel.default
+        # -> quantized_decomposed.dequantize_per_channel.default
+        # We only folding fp32_weight -> quantized_decomposed.quantize_per_channel.default into
+        # int8_weight and leave quantized_decomposed.dequantize_per_channel.default in graph to be fused
```

Comment:
@jerryzh168 isnt this something we were planning to do for quant workflow itself? quant workflow should produce a quantized model with folded weights. Can we do that asap so that changes like this are not required?

Reply:
yeah we can do this in quant flow, but ideally we just rely on constant folding the same way in this PR so we don't reinvent the wheel

Reply:
sure. whatever the mechanism is, it should be part of the quant workflow is what i meant. quant workflow should give you a model with weights quantized.

Reply:
sure, that sounds good, we can do that after this PR